RocketChat/Rocket.Chat
# fix: "Allow Password Change for OAuth Users" setting is not honored in the "Forgot Password" flow

## Proposed changes (including videos or screenshots)

 - Correctly check for the `Accounts_AllowPasswordChangeForOAuthUsers` in the `users.forgotPassword` endpoint.

## Issue(s)


## Steps to test or reproduce

1. Go to **Settings > Accounts > Allow Password Change for OAuth Users** and disable this setting.
2. Create a user via LDAP or OAuth;
3. Log out and use the "Forgot your password?" option to reset the password of the user that has just been created;
**Current behavior:** the reset password e-mail is sent (even to LDAP/OAuth users that didn't have a  password before);
**Expected behavior:** the reset password e-mail should not be sent.


## Further comments

[CORE-439](https://rocketchat.atlassian.net/browse/CORE-439)

[CORE-439]: https://rocketchat.atlassian.net/browse/CORE-439?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
